### ISSUE TITLE
Steam overlay unresponsive/slow

### ISSUE THREAD

**Issue by l2-** [obsproject/obs-studio#5185]
### Operating System Info

Windows 10

### OBS Studio Version

27.0.1

### OBS Studio Log URL

https://obsproject.com/logs/zxydM8Hg7h7Vqzjn

### Expected Behavior

I expect the steam overlay (in game) to work normally and be responsive.

### Current Behavior

The steam overlay is very unresponsive and slow and basically ignores my input very often. This happens for any window part of the steam overlay. When the input is ignored by the overlay the game receives the input instead. ie. when I'm at the home screen of counter strike: global offensive and I have steam overlay opened with a page I can scroll on. If I scroll continuously I can see that some of the scroll is ignored by the steam overlay but still caught by the game where the news page ends up being scrolled.

### Steps to Reproduce

1. Add game capture and turn on use anti-cheat compatibility hook. Set hook rate to fastest for most obvious result.
2. Launch game (in my case Counter Strike: Global Offensive)
3. Open steam overlay
4. Try to interact with the steam overlay such as navigating the browser or typing a message to a friend.
...
Settings i used to reproduce: https://i.imgur.com/XMeETYD.png

### Anything else we should know?

From playing around with settings I think I've narrowed this down to the use anti-cheat compatibility hook. When hook rate is set to slow the input is ignored by steam overlay say every 5-10 seconds consistently (I did not time it exactly). When hook rate is set to fastest the input is ignored by steam overlay almost continuously making it impossible to type a message to a friend using the overlay for example. When game capture is not used or when the use anti-cheat compatibility hook is turned off this problem does not happen and steam overlay is responsive and functions normally. In this case I need the use anti-cheat compatibility hook to capture counter strike while FACEIT anti-cheat is active.

**Comment by Fenrirthviti** [comment]
This reads like a bug report for Steam, not OBS. I don't think there's anything we can do about this one, especially with CS:GO which is already notoriously aggressive against any external hooks.

**Comment by l2-** [comment]
Some more info from testing. I can only reproduce this with csgo i've also tried dark souls 3(dx11) and ac valhalla(dx12) and both do not cause issues for steam overlay. If faceit anti-cheat IS active when csgo is open the overlay works fine. If faceit anti-cheat IS NOT active when csgo is open the overlay is unresponsive and the game capture does not capture csgo, display capture does capture csgo. I've tried this with game capture with capture specific window and capture any full screen window and the problem happens with both modes. It seems to me that use anti-cheat compatibility hook active but unable to hook csgo, when csgo is opened, causes the overlay to be unresponsive but that's just my layman observation.

I guess the work around for me is to keep the csgo specific game capture disabled unless i launch faceit anti-cheat.

**Comment by Fenrirthviti** [comment]
I'll leave this open for someone more familiar with the capture pipelines to weigh in, but CS:GO is definitely a problematic game to capture with Valve's new stance on cheats and how aggressively they block/restrict hooking. There might not be anything we can do.

Keep in mind that the anti-cheat hook is just changing the hook method, and is kind of a misnomer for what it's actually doing. It's called anti-cheat compatibility because the hook method used is less likely to be blocked by anti-cheat, but it shouldn't really be any riskier to leave the option off.

**Comment by RytoEX** [comment]
I cannot reproduce this with OBS Studio 29.0.0 Beta 1.

I added `-allow_third_party_software` to CS:GO's Launch Options (from the game's Properties in Steam). I used the following Game Capture settings in OBS:
* Allow Transparency: Enabled
* Capture Cursor: Enabled
* Use anti-cheat compatibility hook: Enabled
* Capture third-party overlays (such as steam): Enabled
* Hook Rate: Fastest

### CS:GO with FACEIT AC enabled
OBS correctly captures CS:GO and Steam overlays. No slowness or unresponsiveness noticed while interacting with the overlay or with browsers within the overlay.

### CS:GO with FACEIT AC disabled
OBS correctly captures CS:GO and Steam overlays. No slowness or unresponsiveness noticed while interacting with the overlay or with browsers within the overlay.

**Comment by l2-** [comment]
I just did some testing with [OBS Studio 29.0 Beta 1](https://github.com/obsproject/obs-studio/releases/tag/29.0.0-beta1) and the problem still happens for me. Basically when `-allow_third_party_software` is set in launch options everything works correctly but when `-allow_third_party_software` is NOT set in launch options the bug in the original post occurs even with OBS Studio 29.0 Beta 1. The problem I have with setting `-allow_third_party_software` is that I don't want to play matchmaking with this enabled. While it is possible to play matchmaking with `-allow_third_party_software` the game will give you this warning upon searching for a match ![pic](https://i.imgur.com/p9Avexp.png). So far I've simply been using display capture instead since it works for me both with FACEIT AC enabled and disabled. The downsides are basically the difference between game capture and display capture such as being able to control more easily what is captured as well as easily controlling transforms for that capture.

**Comment by RytoEX** [comment]
> I just did some testing with [OBS Studio 29.0 Beta 1](https://github.com/obsproject/obs-studio/releases/tag/29.0.0-beta1) and the problem still happens for me.

Did you use the installer or the zip file? The installer ensures that the game capture DLLs are installed to their correct locations. The zip file does not.

That said, OBS Studio 28.1.2 should produce the same results.

> Basically when `-allow_third_party_software` is set in launch options everything works correctly but when `-allow_third_party_software` is NOT set in launch options the bug in the original post occurs even with OBS Studio 29.0 Beta 1. The problem I have with setting `-allow_third_party_software` is that I don't want to play matchmaking with this enabled. While it is possible to play matchmaking with `-allow_third_party_software` the game will give you this warning upon searching for a match.

I _cannot_ capture CS:GO without `-allow_third_party_software` when launching via Steam. It should not work at all, let alone with responsiveness issues. Furthermore, Valve does not allow this, and we do not support any workflow for capturing CS:GO with Game Capture without `-allow_third_party_software`.

**Comment by l2-** [comment]
> Did you use the installer or the zip file?

The installer.

> I cannot capture CS:GO without -allow_third_party_software when launching via Steam. It should not work at all, let alone with responsiveness issues.

I understand and that's too bad I guess. I mainly made this issue because I used to have game capture and display capture both as active sources so that when I'm playing on faceit the game capture would make sure csgo is capture and when I play matchmaking the display capture would make sure csgo is captured. Since game capture ends up being fully transparent when it cannot hook csgo this allowed me to not have to change anything in obs with regards to playing either matchmaking or faceit. The problem with that is that even when playing matchmaking the game capture wouldn't just do nothing but also cause the issue with the steam overlay.

I understand if this issue should be closed since it is not part of the workflow.
